### ISSUE TITLE
Adding text-to-speech in Streamlit

### ISSUE THREAD

**Issue by programmingninjas** [programmingninjas/GPT-3.5-ON-STEROIDS#3]
Adding text-to-speech feature will improve our frontend and make the experience better.

**Comment by Suraj-kumar00** [comment]
Hi Ayan can I start working on this issue and can you please help me with this like by telling me more about it and giving me some resources and stuff?

**Comment by programmingninjas** [comment]
Sure @Suraj-kumar00 you can work on this. I'm assigning this to you. You have to make the gpt's responses read out loud like it's talking to us. 
[Reference](https://vahidmirjalili.com/blog/posts/chatgpt-voice-assistant/#convert-the-response-to-audio-using-text-to-speech)

**Comment by Suraj-kumar00** [comment]
Thanks, Ayan I'm working on it and can you please tell me like how much time I have to complete this? I started working on this!

**Comment by programmingninjas** [comment]
Don't worry about time. Take it easy, You can take 5-10 days but make sure to tell me if you cannot complete this so that i can assign it to someone else

**Comment by Suraj-kumar00** [comment]
Yeah I'll try my 100% to complete to this task!

**Comment by Suraj-kumar00** [comment]
Hi Ayan, how are you doing?
So I just set up the project and ran it on my local  machine but I encountered an error of this type:
![Screenshot (164)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/0e8fb388-6b18-457e-a318-6e674e547834)


what should i do in this case, by the way I'm working on my issue...

**Comment by programmingninjas** [comment]
You should always refer to the README.md file before working on a project. There i mentioned that you need to get your own API keys for OPENAI and SERP-API. Links are attached there. Kindly go to the respective websites and register to get the keys. Read the code and insert it where I wrote "Your _____ Key"

**Comment by Suraj-kumar00** [comment]
> You should always refer to the README.md file before working on a project. There i mentioned that you need to get your own API keys for OPENAI and SERP-API. Links are attached there. Kindly go to the respective websites and register to get the keys. Read the code and insert it where I wrote "Your _____ Key"

Oh pardon, next time I'll remember and thanks !

**Comment by programmingninjas** [comment]
@Suraj-kumar00 We changed the file structure significantly. So i would advise you to update the fork that you are currently working on. Reach out to me if something is not clear.

**Comment by Suraj-kumar00** [comment]
Yeah @programmingninjas, Like after forked I switched to the feature branch as per the README file and I've up to date that branch, like on GitHub should I also up-to date the main branch? And one thing I want to ask that should I create a draft pull request there we can talk about this if you say I'll create the draft pull request!

**Comment by programmingninjas** [comment]
Yeah update it. And for further communication reach out to me over linkedIn. I guess we are connected right?. [LinkedIn](https://www.linkedin.com/in/ayankhan3105/)

**Comment by Suraj-kumar00** [comment]
> Yeah update it. And for further communication reach out to me over linkedIn. I guess we are connected right?. [LinkedIn](https://www.linkedin.com/in/ayankhan3105/)

yes we are connected, and draft pull request should create it of not ?

**Comment by programmingninjas** [comment]
Nope leave it.

**Comment by Suraj-kumar00** [comment]
Hi @programmingninjas I need help actually when i ad Open AI API key and save it'll automatically disabled the key i tried it 4 to 5 times but I'm getting it can you please help me with this?

**Comment by Suraj-kumar00** [comment]
Oh okay

**Comment by Suraj-kumar00** [comment]
See here this time i never push but still It's showing an error of incorrect API key ??

![Screenshot (175)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/7b6aa30b-9c59-4ed2-a43a-fe71020d53e2)
![Screenshot (176)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/366e5222-e70b-4ec5-b77d-65648a3fb30b)

**Comment by programmingninjas** [comment]
because you are using the old file structure. Clone the current project in a separate folder and you just have to change the keys in the .env file

**Comment by Suraj-kumar00** [comment]
Isn't this will fix using git pull upstream main ?

**Comment by programmingninjas** [comment]
There is a rate limit of 3 requests per minute in this API. The task that you have given the Agent is not a task due to which it's not using any command which takes a min on avg to complete that's why there is a sleep time of just 5 sec between each call as tools itself takes a minute. 
Example tasks:
1) Plan a trip to Japan in August
2) Popular anime of 2023
3) Watch a tutorial and do something with that information
4) Searching about someone like who is programmingninjas on github.

Suraj as this project is in it's early phase there will be many errors along the way. Retrying always help!

**Comment by Suraj-kumar00** [comment]
> There is a rate limit of 3 requests per minute in this API. The task that you have given the Agent is not a task due to which it's not using any command which takes a min on avg to complete that's why there is a sleep time of just 5 sec between each call as tools itself takes a minute. Example tasks:
> 
> 1. Plan a trip to Japan in August
> 2. Popular anime of 2023
> 3. Watch a tutorial and do something with that information
> 4. Searching about someone like who is programmingninjas on github.
> 
> Suraj as this project is in it's early phase there will be many errors along the way. Retrying always help!

thanks @programmingninjas for you help and yeah I'll retry it again.

**Comment by Suraj-kumar00** [comment]
Hi @programmingninjas, How are you doing?
Sorry for being so late here because of some reason but anyway, I'll let you know the progress by tonight...

**Comment by Suraj-kumar00** [comment]
Hello @programmingninjas, How are you doing?
Actually, I'm trying to work on this issue so I take this for 3 hours, but I'm not reaching the end can you please help me? Like the project is working perfectly in my machine everything is fine but the moment I'm trying to convert the text to audio a lot of things happened and I messed up, could you please help me with this? here are the screenshots
![Screenshot (186)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/45c5d9e5-78fd-4d50-bfa4-c160ede624de)
![Screenshot (187)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/db7317a2-f507-47c8-8cb0-56ea1640bf8f)
![Screenshot (188)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/9a210ffa-a704-446c-9f2a-9041ed473957)
![Screenshot (189)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/adc55db0-1738-45b5-b080-e3059b5da025)
![Screenshot (190)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/507a21dd-e6af-49ac-bb57-d3ccc17977d4)
![Screenshot (191)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/e2b89734-01ce-447d-ba62-f0e5e7ba97df)
![Screenshot (192)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/db31d4da-73d9-4df4-8980-3fa8fae54545)
![Screenshot (193)](https://github.com/programmingninjas/GPT-3.5-ON-STEROIDS/assets/123288511/af2160ae-7455-4efe-8550-b4be0ff43894)

**Comment by programmingninjas** [comment]
I'm not able to figure out it out just by the ss. I told you to reach out to me on linkedin bro. We will schedule a gmeet and discuss regarding this. I even texted you there but you didn't reply

**Comment by Suraj-kumar00** [comment]
oh okay

**Comment by RaviRam-Gummuluri** [comment]
Hi, I would like to take up this issue as part of SWOCS4.

**Comment by programmingninjas** [comment]
@RaviRam-Gummuluri sure you can! Kindly set up the project locally and make sure you understand it thoroughly. I will assign this issue to you after 1st Jan as SWOC contributions will start from then. Good Luck!

**Comment by programmingninjas** [comment]
@RaviRam-Gummuluri Assigned you the issue! Start working on them

**Comment by Burhanuddin-2001** [comment]
If this issue is still open, I would like to contribute to it.

**Comment by programmingninjas** [comment]
@RaviRam-Gummuluri are you working on the issue? Else i will assign it to @Burhanuddin-2001
